src/wp-includes/media.php

```diff
@@ -5652,13 +5652,8 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 	 * can result in the first post content image being lazy-loaded or an image further down the page being marked as a
 	 * high priority.
 	 */
-	switch ( $context ) {
-		case 'the_post_thumbnail':
-		case 'wp_get_attachment_image':
-		case 'widget_media_image':
-			if ( doing_filter( 'the_content' ) ) {
-				return $loading_attrs;
-			}
+	if ( 'the_content' !== $context && doing_filter( 'the_content' ) ) {
```

Comment:
Here we should temporarily exclude the `do_shortcode` context still, since as of now it is not handled together with the post content (`the_content`). This is going to change in #4973 (so this is only temporary), but that change should only be made there.

I think once you add this, the failures from the `test_content_rendering_with_shortcodes()` and `test_content_rendering_with_shortcodes_nested()` tests should be resolved.
```suggestion
	// TODO: Handle shortcode images together with the content (see https://core.trac.wordpress.org/ticket/58853).
	if ( 'the_content' !== $context && 'do_shortcode' !== $context && doing_filter( 'the_content' ) ) {
```

Reply:
I will undertake that task. However, I refrained from doing so due to your comment indicating that there was no necessity to update any tests. Nevertheless, I appreciate the additional information you provided.